src/main/java/com/github/manolo8/darkbot/gui/plugins/PluginName.java

```diff
@@ -1,15 +1,31 @@
 package com.github.manolo8.darkbot.gui.plugins;
 
+import com.github.manolo8.darkbot.extensions.plugins.Plugin;
 import com.github.manolo8.darkbot.extensions.plugins.PluginDefinition;
+import com.github.manolo8.darkbot.extensions.plugins.PluginUpdater;
+import com.github.manolo8.darkbot.gui.components.MainButton;
+import com.github.manolo8.darkbot.utils.I18n;
 import net.miginfocom.swing.MigLayout;
 
 import javax.swing.*;
 import java.awt.*;
+import java.awt.event.ActionEvent;
 
 class PluginName extends JPanel {
 
-    PluginName(PluginDefinition definition) {
-        super(new MigLayout("", "[]5px[]5px[]5px[]"));
+    private final Plugin plugin;
+    private final PluginUpdater pluginUpdater;
+
+    private final UpdateButton updateButton;
+
+    PluginName(Plugin plugin, PluginUpdater pluginUpdater) {
```

Comment:
Too much logic for a "name", look at the other comment but essentially keep the button functionality outside, simply pass as parameter the button to display, keep the button as part of the plugin card